state/azure/tablestorage/tablestorage.go

```diff
@@ -38,59 +38,108 @@ Concurrency is supported with ETags according to https://docs.microsoft.com/en-u
 package tablestorage
 
 import (
+	"context"
 	"fmt"
 	"strings"
+	"time"
 
-	"github.com/Azure/azure-sdk-for-go/storage"
-	"github.com/agrea/ptr"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/aztables"
 	jsoniter "github.com/json-iterator/go"
 	"github.com/pkg/errors"
 
+	azauth "github.com/dapr/components-contrib/authentication/azure"
 	"github.com/dapr/components-contrib/state"
 	"github.com/dapr/kit/logger"
 )
 
 const (
 	keyDelimiter        = "||"
 	valueEntityProperty = "Value"
-	operationTimeout    = 1000
 
-	accountNameKey = "accountName"
-	accountKeyKey  = "accountKey"
-	tableNameKey   = "tableName"
+	accountNameKey  = "accountName"
+	accountKeyKey   = "accountKey"
+	tableNameKey    = "tableName"
+	cosmosDbModeKey = "cosmosDbMode"
```

Comment:
This will never support other modes. The upstream SDK only can work with two types of table storage. We want the current default to be storage accounts to avoid a breaking change.

A toggle between CosmosDB and storage accounts is all that is necessary.

I really find this solution better as a user given this will never support other table stores.